### ISSUE TITLE
Script is spamming WH on incorrect levels?

### ISSUE THREAD

**Issue by JeanRotmg** [wchill/steamSummerMinigame#335]
Using the "tracking feature" i found myself using WH on other rooms than x00.
I know its supposed to spam WH before we reach 100k but i am way after 100k (near 90M).

My script is up to date (even regarding silent updates, page refreshed), i dont do anything , i dont have other scripts besides autoupgrade (which is also up to date)

Is this a normal behavior? if so is it planned to fix the tracking feature? Or is it just lag?

Also , can you add a warning about minimizing/tab, mixing scripts, on the first launch?

**Comment by TheSnowQueen** [comment]
If you have Anity-troll he will spam.
Travist say this : "if you have far too many WH or LN than you can spend, it will spam them"
Btw it's will be nice to stop this spam when you are near x00

**Comment by MikeZeDev** [comment]
Thanks, but i forgot to mention that it wasnt anti troll  :)
Seems to be fixed with 7.0 anyway, we wiil see

**Comment by K4r0qtuYNE5G4qgZ** [comment]
7.0.2 Spams non stop at non x00 burns thousands is that normal? I need to refresh to stop it from doing for some time, then starts again, also misses x00 becauee of it?!

**Comment by DrWindu** [comment]
I second this issue - 7.0.3 spams for me too. I'm not a JS dev so I can't debug properly in console, but I would suggest looking quickly into the "wormHoleConstantUseOverride" and "wormHoleConstantUse" variables - they switch to true whenever one has many WH, and it may be fucking up 100M players trying to help, turning them into random spammers.

It looks like an attempt to brute-force a room up when reset gets too close, but I believe it wastes much time by reducing odds of hitting x00 levels. I saw myself using WH on 99 levels, which i'm pretty sure is NOT intended...

**Comment by K4r0qtuYNE5G4qgZ** [comment]
I will try to tweak it live it burns way to much currently i will be low before 10m for sure and started with 50k WH and 50k LN it burns thousands on non x00 Levels...

**Comment by DrWindu** [comment]
TL;DR : The script spams LN and WH inefficiently when the deadline closes, which I'm not sure is a sound strategy. There is also a different bug for me, which may be very low priority because of my setup.
[Sorry for verbosity, i'm trying to give the details.]

Okay, so I did some more debugging and the spam I'm seing is NOT from being too close to reset time (which I still think is a dubious feature, but I'm assuming Trouvist knew what he did with commit 018afd2).

I've tried to add console logs to main loop by hand, but they all stop mysteriously when the script starts acting up and firing WH/LN. Around the same time, I start getting errors saying "TypeError: w.BigNumber is not a constructor" - which may be unrelated.

Then all that's left is "Got stale data" ad lib, and my wormholes just keep firing. I'm not much of a JS dev, so I suspect the MainLoop has crashed or otherwise stopped, and an aggressive callback is still running, firing abilites... but I'm not actually sure.

I've been told that it may be my setup (Iceweasel on Debian32, which is a rebrand of Firefox 31.7.0, probably with exact same code except for the name and logo, for branding issues) - if so, my bug is probably low priority.

I still believe, however, that the constantWormHoleUseOverride thing is a bug, which should be fixed.

**Comment by K4r0qtuYNE5G4qgZ** [comment]
i have the same and run it on Chrome 64bit...

**Comment by K4r0qtuYNE5G4qgZ** [comment]
analyzed it!! doesnt seem to happen if you buy all abilities, but doesn't auto buy, serious bug if someone don't buy all, script freaks out and fires wh non stop...

**Comment by wchill** [comment]
Calm down, we are looking into it. I'm consulting with @Trouvist right now.

**Comment by CrashCat** [comment]
I guess #353 is a dupe of this.  Any word on what to do?  I'm going to refrain until we have some definite answer because while I don't mind dumping 60k wormholes I don't want to grief a room by blowing past 00 levels constantly.

**Comment by CrashCat** [comment]
To confirm I tried latest SteamDB script (after disabling the others) and it is not WH dumping except on 00 levels.  I think something in my loadout for your script is broken but I can't tell what.

**Comment by SpoilSpot** [comment]
If your wormhole count > 3 * seconds left, then the constantWormholeOverride gets set.
That may still not be an efficient use of wormholes if you keep missing the x00 levels.
There's currently ~14000 seconds left, and I have >50k wormholes, and so does a lot of other people, so it's almost impossible to hit x00 levels, even without anyone deliberately sabotaging it.

**Comment by HiRusSai** [comment]
^ True, fix pls

**Comment by DrWindu** [comment]
I believe we can mark this as fixed after @Trouvist 's latest update, as well as 353.
